client/python/gradio_client/client.py

```diff
@@ -603,6 +605,137 @@ def _get_config(self) -> dict:
+            api_names: The api_names of the app to turn into bot commands. This parameter currently has no effect as ChatInterface only has one api_name ('/chat').
+            command_names: The names of the command the bot will have. Must be the same length as api_names. For gr.ChatInterface, this list must contain one element. For example, if the value is ['bot'], the bot command can be triggered via !bot.
+            to_id: The name of the space hosting the discord bot. If None, the name will be gradio-discord-bot-{random-substring}
+            hf_token: HF api token with write priviledges in order to upload the files to HF space. Can be ommitted if logged in via the HuggingFace CLI, unless the upstream space is private. Obtain from: https://huggingface.co/settings/token
+            private: Whether the space hosting the discord bot is private. The visibility of the discord bot itself is set via the discord website. See https://huggingface.co/spaces/freddyaboulton/test-discord-bot-v1
+            persist_state: Currently has no effect.
+        """
+        if self.config["mode"] != "chat_interface":
+            raise ValueError(
+                "Only demos built with gr.ChatInterface can be deployed as discord bots automatically for now."
+            )
+
+        if not command_names:
+            command_names = ["chat"]
+
+        assert (
+            len(command_names) == 1
+        ), f"gr.ChatInterface can only accept one command name. Received {', '.join(command_names)}"
+        api_names = ["chat"]
+
+        is_private = False
+        if self.space_id:
+            is_private = huggingface_hub.space_info(self.space_id).private
+            if is_private:
+                assert hf_token, (
+                    f"Since {self.space_id} is private, you must explicitly pass in hf_token "
+                    "so that it can be added as a secret in the discord bot space."
+                )
+
+        if to_id:
+            if "/" in to_id:
+                to_id = to_id.split("/")[1]
+            space_id = huggingface_hub.get_full_repo_name(to_id, token=hf_token)
+        else:
+            if self.space_id:
+                space_id = f'{self.space_id.split("/")[1]}-gradio-discord-bot'
+            else:
+                space_id = f"gradio-discord-bot-{secrets.token_hex(4)}"
+            space_id = huggingface_hub.get_full_repo_name(space_id, token=hf_token)
+
+        api = huggingface_hub.HfApi()
+
+        try:
+            huggingface_hub.space_info(space_id)
+            first_upload = False
+        except huggingface_hub.utils.RepositoryNotFoundError:
+            first_upload = True
+
+        huggingface_hub.create_repo(
```

Comment:
Is doing this really needed? I think you can use `huggingface_hub.duplicate_space` and let it handle the duplication